packages/nimble-components/src/wafer-map/types.ts

```diff
@@ -79,3 +87,12 @@ export interface WaferMapValidity extends ValidityObject {
     readonly invalidGridDimensions: boolean;
     readonly invalidDiesTableSchema: boolean;
 }
+
+export interface WaferRequiredTypeMap {
+    [key: string]: DataType;
+    [key: number]: DataType;
+    [key: symbol]: DataType;
+    colIndex: Int32;
+    rowIndex: Int32;
+    value: Float64;
+}
```

Comment:
I sunk it down and played with it locally and I don't think we actually need to specify the index signatures if we do the following (I recommend going this way comments and all):

```suggestion
// Apache arrow probably should not be using a Record and index types on TypeMap
// because in strict checking they end up required.
// See: https://github.com/apache/arrow/issues/12663#issuecomment-1088244575
// We can work around that issue by using a type alias instead of an interface
// Where index signatures are looser.
// See: https://github.com/microsoft/TypeScript/issues/15300#issuecomment-1317901527
// eslint-disable-next-line @typescript-eslint/consistent-type-definitions
export type WaferRequiredFields = {
    colIndex: Int32,
    rowIndex: Int32,
    value: Float64
};
```